### PR TITLE
Make log_server setting work again and remove log_payloads

### PR DESCRIPTION
**Pull request by rchl**
This setting was not used anywhere, instead log_debug controlled logging
in the panel.

Change default to False and make it control the log panel logging.

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## Self-help instructions
 
-Enable LSP logging: In LSP Settings enable the `log_debug` setting and `log_payloads` if needed.
+Enable LSP logging: In LSP Settings enable the `log_debug` setting.
 Enable server logging: set `log_server` and `log_stderr` to `true`
 Restart Sublime and open the console (``ctrl+` ``) to see the additional logging.
```
Comment (rwols) [inline docs/troubleshooting.md:5]: This should probably say: Run LSP: Toggle Log Panel from the command palette. No restart is needed.

File: plugin/core/windows.py [hunk 2]
```diff
@@ -666,7 +666,7 @@ def __init__(self, manager: WindowManager, server_name: str) -> None:
         self._manager = ref(manager)
         self._server_name = server_name
 
-    def log(self, message: str, params: Any, log_payload: bool) -> None:
+    def log(self, message: str, params: Any, log_payload: Optional[bool] = True) -> None:
```
Comment (rwols) [inline plugin/core/windows.py:669]: The type of log_payloads should be `bool` and its default value should be `False`.

Comment (rchl) [reply plugin/core/windows.py:669]: Why default `False`? Then we have to pass `True` in most cases where now we can just skip it.
Do you want thing more explicit?

Comment (rwols) [reply plugin/core/windows.py:669]: Right, sorry, I think I need more coffee. I only have a problem with the Optional. That makes the type have 3 values whereas 2 possible values should be sufficient. The default should stay `True`.

**Pull request merged**